brave/brave-core
# Android private search fix

Resolves brave/brave-browser/issues/25821
Resolves brave/brave-browser/issues/26483

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

### test 1 (the issue)

1. Open a private tab.
2. Search for something.
3. Change the default search engine for private tabs (Settings -> Search engines -> Private Tab).
4. Navigate back to existing private tab OR open a new private tab.
5. Search for something, new search engine shall be used.

### test 2 (upgrading from and old version)

1. When upgraded from old version, the search settings for regular and private tab shall remain then same as before.

### test 3 (private tabs initialization)

1. Clear all tabs.
2. Open a private tab. Tap on url bar and search anything.
3. Clear private tabs and go back to a regular tab.
4. Open a private tab and search in the url bar again.
5. Repeat for 2-3 times.
6. Nothing extraordinary shall happen during the process.

### test 4 (onboarding - non-Brave default)

1. Set Android locale to anywhere Brave Search IS NOT default, reset/reinstall Brave.
2. Get through onboarding and tap url bar twice to get the search onboarding fragment.
3. Choose any one of the search options.
4. Check in settings that search settings for both regular and private tabs are set to the search engine that was just being chosen.


### test 4 (onboarding - Brave default)

1. Set Android locale to anywhere Brave Search IS default, reset/reinstall Brave.
2. Check in settings that Brave search is set for both regular and private tabs.
